tests/test_modulemd_depsolver.py

```diff
@@ -0,0 +1,291 @@
+            ModulemdDependency(name="dependency_1", stream="11.1"),
+        ],
+    )
+
+    excluded_module = ModulemdUnit(
+        name="excluded",
+        stream="6.30",
+        version=3,
+        context="ABC",
+        arch="x86_64",
+        artifacts=[
+            "excluded-archive-tar-0:2.32-440.module+el8.3.0+6718+7f269185.src",
+            "excluded-archive-zip-0:1.67-1.module+el8.3.0+6718+7f269185.noarch",
+        ],
+    )
+
+    module_dep_1 = ModulemdUnit(
+        name="dependency_1",
+        stream="11.1",
+        version=1,
+        context="def",
+        arch="x86_64",
+        artifacts=[
+            "dep1-4:5.30.1-452.module+el8.4.0+8990+01326e37.src",
+            "dep1-4:5.30.1-452.module+el8.4.0+8990+01326e37.x86_64",
+            "dep1-archive-tar-0:2.32-440.module+el8.3.0+6718+7f269185.src",
+            "dep1-archive-zip-0:1.67-1.module+el8.3.0+6718+7f269185.noarch",
+        ],
+    )
+
+    module_dep_2 = ModulemdUnit(
+        name="dependency_2",
+        stream="2.22",
+        version=2,
+        context="efg",
+        arch="x86_64",
+        artifacts=[
+            "dep1-4:5.30.1-452.module+el8.4.0+8990+01326e37.src",
+            "dep1-4:5.30.1-452.module+el8.4.0+8990+01326e37.x86_64",
+            "dep1-archive-tar-0:2.32-440.module+el8.3.0+6718+7f269185.src",
+            "dep1-archive-zip-0:1.67-1.module+el8.3.0+6718+7f269185.noarch",
+        ],
+    )
+
+    repo_1_units = [module1, module2, module_dep_1, module_dep_2, excluded_module]
+    pulp.insert_units(repo_1, repo_1_units)
+
+    # Define mock module units for repo_2
+    module4 = ModulemdUnit(
+        name="module_in_second_repo",
+        stream="8.30",
+        version=6,
+        context="fgh",
+        arch="x86_64",
+        artifacts=[
+            "secondRepoRPM-4:5.30.1-452.module+el8.4.0+8990+01326e37.src",
+            "secondRepoRPM-4:5.30.1-452.module+el8.4.0+8990+01326e37.x86_64",
+        ],
+        dependencies=[
+            ModulemdDependency(
+                name="dependency_2", stream="something completely different"
+            ),
+            ModulemdDependency(name="dependency_1", stream="11.1"),
+        ],
+    )
+
+    module_dep_3 = ModulemdUnit(
+        name="dependency_2",
+        stream="something completely different",
+        version=3,
+        context="efg",
+        arch="x86_64",
+        artifacts=[
+            "dep1-4:5.30.1-452.module+el8.4.0+8990+01326e37.src",
+            "dep1-4:5.30.1-452.module+el8.4.0+8990+01326e37.x86_64",
+            "dep1-archive-tar-0:2.32-440.module+el8.3.0+6718+7f269185.src",
+            "dep1-archive-zip-0:1.67-1.module+el8.3.0+6718+7f269185.noarch",
+        ],
+    )
+
+    repo_2_units = [module4, module_dep_3]
+    pulp.insert_units(repo_2, repo_2_units)
+
+    expected_output_set = list(
+        [
+            (module.name, module.stream, repo_1.id)
+            for module in repo_1_units
+            if not module.name == "excluded"
```

Comment:
This might deserve a better name :)
We don't actively exclude any modules but rather we include only those defined in ubi-config.

Reply:
Replaced with "ignored"